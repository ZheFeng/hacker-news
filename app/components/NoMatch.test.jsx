/* eslint-disable import/no-extraneous-dependencies, max-nested-callbacks */
/* global describe, it, expect */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import NoMatch from './NoMatch';

const location = {
  pathname: 'test',
};

describe('<NoMatch />', () => {
  it('Snapshot Test', () => {
    const component = renderer.create(
      <NoMatch location={location} />,
  );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('show missed warning message', () => {
  // Render a checkbox with label in the document
    const noMatch = shallow(<NoMatch location={location} />);
    const message = `Sorry but ${location.pathname} didnâ€™t match any pages`;
    expect(noMatch.find('p').text()).toEqual(message);
  });
});
