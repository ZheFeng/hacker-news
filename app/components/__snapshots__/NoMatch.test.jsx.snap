exports[`<NoMatch /> Snapshot Test 1`] = `
<div>
  <h2>
    Whoops
  </h2>
  <p>
    Sorry but 
    test
     didnâ€™t match any pages
  </p>
</div>
`;
